### PR TITLE
Add more e2e tests

### PR DESCRIPTION
**Pull request by artonge**
Fix https://github.com/nextcloud/photos/issues/1322

## Added tests

- Test creating, accessing, and deleting public links
- Test adding files to a shared album from timeline
- Test multiple collaborators for shared albums
    - See each other pictures
    - Removing a collaborator removes its pictures 
    - Can remove each other pictures
- Testing that deleting a user removes it from the collaborators list if albums
- Testing that deleting a user deletes its albums
- Test if deleting file break an album

## Enhancements

- Use `NcSelect` for collaborator selection form
- Support browsers that do not support caches
- Add jest's types for tests files
- Improve resilience of 2e2 tests

## Fixes

- Prevent deletion of a received album when it is received from group share
- List received shared albums in album picker
- Remove user from collaborators when it is deleted

### CODE REVIEW

**Comment by artonge** [comment]
Screenshot from 2023-05-24 14-53-41

@Pytal I remember you add the same issue some time ago, is there a fix?
Ideally, I would like the input not to be focused when the pop-up appears.

Edit:
I am using this hack for now. Better idea welcome :)
```js
:dropdown-should-open="({open, search}) => open && search !== ''"
```

**Comment by Pytal** [comment]
🧐 was fixed in https://github.com/nextcloud-deps/vue-select/pull/14

Before the fix this happened because the dropdown itself would overflow the container but in this case it doesn't look like it would overflow, perhaps the fix above though may provide a hint to the bug in this specific case @artonge?

**Comment by Pytal** [comment]
> Edit: I am using this hack for now. Better idea welcome :)
> 
> ```js
> :dropdown-should-open="({open, search}) => open && search !== ''"
> ```

If the dropdown never needs to be placed above the input then `:append-to-body="false"` should work @artonge

**Review by AndyScherzinger — APPROVED** [review]

File: cypress.config.ts [hunk 1]
```diff
@@ -17,9 +17,11 @@ export default defineConfig({
 	viewportWidth: 1280,
 	viewportHeight: 720,
 
+	requestTimeout: 20000,
+
 	// Tries again 2 more times on failure
```
Comment (Pytal) [inline cypress.config.ts:22]:
```suggestion
```
```
runMode: 0,
```

**Review by Pytal — APPROVED** [review]

**Comment by skjnldsv** [comment]
You're close to success it seems!!

**Comment by artonge** [comment]
/backport to stable27 1d8058a50c4c78f635e6c4d511f6e7bbc28c7233

**Comment by artonge** [comment]
/backport 1d8058a50c4c78f635e6c4d511f6e7bbc28c7233to stable27

**Comment by artonge** [comment]
/backport 1d8058a50c4c78f635e6c4d511f6e7bbc28c7233to to stable27

**Pull request merged**